hack/jenkins/test-history/gen_html.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Creates an HTML report for all jobs starting with a given prefix.
+
+Reads the JSON from tests.json, and prints the HTML to stdout.
+
+This code is pretty nasty, but gets the job done.
+
+It would be really spiffy if this used an HTML template system, but for now
+we're old-fashioned. We could also generate these with JS, directly from the
+JSON. That would allow custom filtering and stuff like that.
+"""
+
+from __future__ import print_function
+
+import json
+import string
+import sys
+
+def gen_tests(data, prefix):
+    """Creates the HTML for all test cases.
+
+    Returns the HTML along with the number of passing, unstable, failing, and
+    skipped tests.
+    """
+    html = '<ul class="test">\n'
```

Comment:
FYI, I tend to avoid +=ing string like this in favor of a generator:

def Parts():
  yield '<ul class="test">\n'
  for foo in bar: yield '<li>%s</li>' % foo
  yield '</ul>'

return ''.join(str(s) for s in Parts())